### PR TITLE
Backup and restore now supports the period, comma, tab, newline, and exclamation point.

### PR DESCRIPTION
**Pull request by khuddlefish**
We now support the above special characters in schema and table naming for backup and restore operations to be more in line with postgres functionality.

Authors: Karen Huddleston and Jamie McAtamney

### CODE REVIEW

File: gpMgmt/bin/gppylib/operations/test/unit/test_unit_backup_utils.py [hunk 1]
```diff
@@ -1141,3 +1125,272 @@ def test_init_context_with_no_mdd(self):
...
+        results = list_to_csv_string([])
+        self.assertEqual('\n', results)
+
+    def test_list_to_csv_string_special_chars(self):
+        table_list = ['public', 'foo!$"\t\n,.']
+        expected = 'public,"foo!$""\t\n,."\n'
+        results = list_to_csv_string(table_list)
+        self.assertEqual(expected, results)
+
+    def test_list_to_csv_string_special_chars_no_delimiter_or_terminator_or_quotechar(self):
```
Comment (kaknikhil) [inline gpMgmt/bin/gppylib/operations/test/unit/test_unit_backup_utils.py]: maybe add a test for list_to_csv_string when the first item (i.e. schema) in the list has special characters.

File: gpMgmt/bin/gppylib/operations/backup_utils.py [hunk 2]
```diff
@@ -454,6 +468,64 @@ def write_lines_to_file(filename, lines):
...
+        if row and len(row) != 2:
+            raise Exception("%s is not in the format schema.table" % ".".join(row))
+        content.append(row)
+    return content
+
+def list_to_csv_string(items, delimiter=',', terminator='\n'):
+    csv_string = StringIO.StringIO()
+    writer = csv.writer(csv_string, delimiter=delimiter, quotechar='"', lineterminator=terminator)
+    writer.writerow(items)
+    return csv_string.getvalue()
```
Comment (kaknikhil) [inline gpMgmt/bin/gppylib/operations/backup_utils.py]: csv_string.close()?

File: gpAux/client [hunk 3]
```diff
@@ -1 +0,0 @@
-Subproject commit 14f1c6c3d966990d3bc0e7d52bf61363b197bf21
```
Comment (Chibin) [inline gpAux/client]: We need to remove this change.

**Comment by Chibin** [comment]
Closing this PR. There will be a follow up PR with new more significant changes.

**Pull request closed without merging**